src/services/uniswapV3/UniswapV3Service.ts

```diff
@@ -0,0 +1,264 @@
+import {injectable} from 'inversify';
+import {Token, CurrencyAmount} from '@uniswap/sdk-core';
+import {FeeAmount, Pool, TickMath, tickToPrice} from '@uniswap/v3-sdk';
+import JSBI from 'jsbi';
+
+import {GraphTick, BarChartTick, TickProcessed} from './interfaces.js';
+import axios from 'axios';
+
+enum Direction {
+  ASC,
+  DESC,
+}
+
+const MAX_INT128 = JSBI.subtract(JSBI.exponentiate(JSBI.BigInt(2), JSBI.BigInt(128)), JSBI.BigInt(1));
+
+@injectable()
+export default class UniswapV3Service {
+  async getFullTickData(poolAddress: string): Promise<GraphTick[]> {
+    let allTicks: GraphTick[] = [];
+    let skip = 0;
+    let loadingTicks = true;
+
+    while (loadingTicks) {
+      const ticks = await this.getTickDataFromSubgraph(poolAddress, skip);
+      allTicks = allTicks.concat(ticks);
+      if (ticks.length < 1000) {
+        loadingTicks = false;
+      } else {
+        skip += 1000;
+      }
+    }
+
+    return allTicks;
+  }
+
+  async getTickDataFromSubgraph(poolAddress: string, skip: number): Promise<GraphTick[]> {
```

Comment:
In case you prefer, in `services/graph` there is a `GraphClient` that could be used. Then there would be more abstraction for this service and it could be unit tested easier through dependency injection.